app/forms/webauthn_verification_form.rb

```diff
@@ -99,16 +126,65 @@ def public_key
     webauthn_configuration&.credential_public_key
   end
 
-  def extra_analytics_attributes
-    auth_method = if webauthn_configuration&.platform_authenticator
-                    'webauthn_platform'
-                  else
-                    'webauthn'
-                  end
+  def generic_error_message
+    if platform_authenticator?
+      t(
+        'two_factor_authentication.webauthn_error.try_again',
+        link: link_to(
+          t('two_factor_authentication.webauthn_error.additional_methods_link'),
+          login_two_factor_options_path,
+        ),
+      )
+    else
+      t(
+        'two_factor_authentication.webauthn_error.connect_html',
+        link_html: link_to(
+          t('two_factor_authentication.webauthn_error.additional_methods_link'),
+          login_two_factor_options_path,
+        ),
+      )
+    end
+  end
+
+  def screen_lock_error_message
+    if user_has_other_authentication_method?
+      t(
+        'two_factor_authentication.webauthn_error.screen_lock_other_mfa',
+        link_html: link_to(
+          t('two_factor_authentication.webauthn_error.use_another_method'),
+          login_two_factor_options_path,
+        ),
+      )
+    else
+      t(
+        'two_factor_authentication.webauthn_error.screen_lock_no_other_mfa',
+        link_html: link_to(
+          t('two_factor_authentication.webauthn_error.use_another_method'),
+          login_two_factor_options_path,
+        ),
+      )
+    end
+  end
 
+  def extra_analytics_attributes
     {
       multi_factor_auth_method: auth_method,
       webauthn_configuration_id: webauthn_configuration&.id,
     }
   end
+
+  def auth_method
+    if platform_authenticator?
+      'webauthn_platform'
+    else
+      'webauthn'
+    end
+  end
+
+  def user_has_other_authentication_method?
+    MfaContext.new(user).two_factor_configurations.any? do |method|
+      !method.is_a?(WebauthnConfiguration) ||
+        method.platform_authenticator? != platform_authenticator?
+    end
```

Comment:
calling it `method` tells me it would be an instance of `Method`, WDYT of `configuration` ?
```suggestion
    MfaContext.new(user).two_factor_configurations.any? do |configuration|
      !configuration.is_a?(WebauthnConfiguration) ||
        configuration.platform_authenticator? != platform_authenticator?
    end
```

Reply:
Yeah, I think `configuration` is better here 👍